src/query/expression/src/function.rs

```diff
@@ -604,24 +605,20 @@ impl<'a> EvalContext<'a> {
         }
         match &self.errors {
             Some((valids, error)) => {
-                let first_error_row = if let Some(selection) = selection {
-                    if let Some(first_invalid) =
-                        selection.iter().find(|idx| !valids.get(**idx as usize))
-                    {
+                let first_error_row = match selection {
+                    None => valids.iter().enumerate().find(|(_, v)| !v).unwrap().0,
+                    Some(_) if valids.len() == 1 => 0,
```

Comment:
In what case `valids.len() == 1`?

Reply:
The max index of selection should not be larger than num_rows. It seems to be a bug elsewhere.

Reply:
Done.